### DISCUSSION TITLE
New rule: Maintaining parameter order

### DISCUSSION THREAD

**Discussion by PaulCombal** [PHP-CS-Fixer/PHP-CS-Fixer#8208] [Feature ideas]
Assuming the following function:

```php
function foo($a, $b) {
  return $a + $b;
}
```

I would really like a rule that keeps the named parameters in the same order than the function/method signature.

Eg: Replacing the following call:

```php
foo(b: 2, a: 4);
```

With the ordered parameters as such:

```php
foo(a: 4, b: 2);
```

**Comment by kubawerlos** [comment]
There will not be much value in such a fixer.

Fixers work in a per-file context, so it is impossible to fix the call if the function is defined in a different file.

**Reply by realjjaveweb** [reply]
Oh, there _**will**_ be much value in such fixer, however you're right, this would require fixer to do a deep static analysis (like PHPStan) - so in theory possible, but may be harder than it looks.

@PaulCombal try to find/create a [Rector](https://github.com/rectorphp/rector) rule that would do this.

**Reply by kubawerlos** [reply]
Let me rephrase: there will be much value in such automatic re-ordering, while not in the fixer, as it will be limited to fixing only in files where the function is defined.

However, this is totally possible with the help of reflection. During analysis, when named arguments are found, it would not be hard to calculate the FQCN, if any, and call `ReflectionMethod`/`ReflectionFunction` to check the other. That would not be static analysis anymore (meaning a custom fixer, I guess), but would work.
